lib/matplotlib/path.py

```diff
@@ -206,12 +206,13 @@ def _fast_from_codes_and_verts(cls, verts, codes, internals=None):
         return pth
 
     def _update_values(self):
+        self._simplify_threshold = rcParams['path.simplify_threshold']
         self._should_simplify = (
             rcParams['path.simplify'] and
-            (len(self._vertices) >= 128 and
-             (self._codes is None or np.all(self._codes <= Path.LINETO)))
+            len(self._vertices) >= 128 and
+            (self._codes is None or np.all(self._codes <= Path.LINETO)) and
+            self._simplify_threshold > 0
```

Comment:
the test on simplify_threshold should go above (say just after path.simplify), in order to avoid iterating on the codes if possible?  (yes, I realize that's unlikely to be the bottleneck)

also it looks like we may be able to deprecate path.simplify and fold it into path.simplify_threshold being zero or not.